src/mantarray_desktop_app/mc_comm.py

```diff
@@ -243,39 +258,33 @@ def _process_next_communication_from_main(self) -> None:
             raise UnrecognizedCommandFromMainToMcCommError(
                 f"Invalid communication_type: {communication_type}"
             )
-        self._command_awaiting_response = comm_from_main
+        self._commands_awaiting_response.append(comm_from_main)
+        # Tanner (3/23/21): consider replacing this with is_queue_eventually_empty
         if not input_queue.empty():
             self._process_can_be_soft_stopped = False
 
-    def _process_main_module_comm(self, comm_from_instrument: bytes) -> None:
-        packet_body = comm_from_instrument[
-            SERIAL_COMM_ADDITIONAL_BYTES_INDEX:-SERIAL_COMM_CHECKSUM_LENGTH_BYTES
-        ]
-
-        packet_type = comm_from_instrument[SERIAL_COMM_PACKET_TYPE_INDEX]
-        if packet_type == SERIAL_COMM_CHECKSUM_FAILURE_PACKET_TYPE:
-            returned_packet = SERIAL_COMM_MAGIC_WORD_BYTES + packet_body
-            raise SerialCommIncorrectChecksumFromPCError(returned_packet)
-
-        if packet_type == SERIAL_COMM_STATUS_BEACON_PACKET_TYPE:
-            pass  # TODO Tanner (3/17/21): Implement this in a story dedicated to parsing/handling errors codes
-        elif packet_type == SERIAL_COMM_SIMPLE_COMMAND_PACKET_TYPE:
-            if self._command_awaiting_response is None:
-                raise NotImplementedError(  # stop mypy complaining
-                    "_command_awaiting_response shoudn't be None here"
-                )
-            prev_command = self._command_awaiting_response
-            if prev_command["command"] == "get_metadata":
-                prev_command["metadata"] = parse_metadata_bytes(packet_body)
-            self._board_queues[0][1].put_nowait(
-                prev_command
-            )  # Tanner (3/17/21): to be consistent with OkComm, command responses will be sent back to main after the command is acknowledged by the Mantarray
-            self._command_awaiting_response = None
-        else:
-            module_id = comm_from_instrument[SERIAL_COMM_MODULE_ID_INDEX]
-            raise UnrecognizedSerialCommPacketTypeError(
-                f"Packet Type ID: {packet_type} is not defined for Module ID: {module_id}"
-            )
+    def _handle_handshake(self) -> None:
+        board_idx = 0
+        board = self._board_connections[board_idx]
```

Comment:
since `board` isn't used in this function except for the if statement could it be
```python
if not self._board_connections[board_idx]:
    return
```

also is there ever a chance that `self._board_connections` is `None`? If so this will raise.

Reply:
`self._board_connections` will always be a list, but it's items can be `None`

Reply:
Also I'm not sure the exact cases we do this, but we've typically been using `if x is None:` instead of `if not x:` to prevent weird edge cases where the variable we're testing is falsey but not actually `None`. I don't think that necessarily applies to an instance of `MantarrayMcSimulator`, but I wrote the check that way to be consistent with how `ok_comm` checks for board connections

Reply:
and it's always going to have at least 1 element?

Reply:
It will have exactly the number of maximum allowed board connections which is currently 4. We don't actually support connecting to 4 instruments at the moment (and possibly never will), this is just a placeholder for down the road when that may need to be supported

Reply:
👍
I just meant that `board` is only used to check if it's `None` or not so you could write it without the unused variable instead, it's up to you though.